tsdb/mapper.go

```diff
@@ -16,8 +16,9 @@ import (
 // aggregate output Time is ignored, and actual Time-Value pairs are contained soley
 // within the Value field.
 type MapperValue struct {
-	Time  int64       `json:"time,omitempty"`  // Ignored for aggregate output.
-	Value interface{} `json:"value,omitempty"` // For aggregate, contains interval time multiple values.
+	Time  int64             `json:"time,omitempty"`  // Ignored for aggregate output.
+	Value interface{}       `json:"value,omitempty"` // For aggregate, contains interval time multiple values.
+	Tags  map[string]string `json:"tags,omitempty"`  // Meta tags for results
```

Comment:
`TagSets` are for grouping the results by dimensions.  For instance, in the following data:

```
cpu,region=us-east,host=A value=1
cpu,region=us-east,host=B value=2
cpu,region=us-west,host=B value=3
cpu,region=us-east,host=A value=4
```

This select statement:

`select * from cpu group by *`
creates 3 tag sets:

```
cpu,region=us-east,host=A
cpu,region=us-east,host=B
cpu,region=us-west,host=B
```

And will group the results accordingly.

If I issue this query though:

`select * from cpu`

There are no tagsets at all, as there are no groups. You actually just get one tagset, with no tags, and 3 series.

That's why we return the tags in the MapperValues now, as they are per point, not per tagset (which may or may not have any actual tags populated).